client/setec/store.go

```diff
@@ -227,10 +253,12 @@ func (s *Store) secretLocked(name string) Secret {
 	f, ok := s.active.f[name]
 	if !ok {
 		f = func() []byte {
-			s.active.RLock()
-			defer s.active.RUnlock()
+			s.active.Lock()
+			defer s.active.Unlock()
 			s.countSecretFetch.Add(1)
-			return s.active.m[name].Value
+			cs := s.active.m[name]
+			cs.LastAccess = s.timeNow()
```

Comment:
afaict this is the only use of timeNow() that needs unix time, so imo it would be cleaner if timeNow() returned time.Time (as is conventional in Go for time) and this one site did s.timeNow().Unix().

Reply:
There are a few, but I agree it's a little weird to force both sides of the boundary to track the seconds. Updated.